### ISSUE TITLE
Controls/Layout does not draw controls correctly in first-time view.

### ISSUE THREAD

**Issue by SecureSharing** [dotnet/maui#4672]
### Description

I am writing a POC MAUI app and feel this behavior is a bug.

In my app, I have created a desktop shell which is basically copied from the Podcasts example code. My app has four pages with some text and frames. When I run the app and view a page for the first time, the controls are not drawn properly. Alignment is off. Margins are incorrect. The page looks like a mess. If I go to another page and return to the previous page, the layout is drawn correctly.

One thing I observed is that this behavior only exists if the layout is contained within a Frame. I was testing a MAUI control from Syncfusion and it drew correctly the first time through when the control was not wrapped inside a frame.  When I added the frame around the control, it did NOT draw correctly. I first thought this may be an issue with the Syncfusion control itself, but then on another page, I had a frame that simply contained two labels. The text was all incorrectly drawn on the first visit to the page, but then drawn correctly on the second visit so I am taking Syncfusion off the hook. I am pretty certain it has something to do with the way the Frame performs its layout calculations. Also, when I resized the window, the layout is recalculated on the size event, and it draws correctly.

Here are some example screenshots:

![First time in with frame](https://user-images.githubusercontent.com/60759137/153909424-c3f3b92a-639b-4939-9075-b188b37090f1.png)
First time in with frame

![Second time in with frame](https://user-images.githubusercontent.com/60759137/153909525-f67e6143-c899-4f2d-9407-69ef1e06ed94.png)
Return to page with frame

![First time in with no frame](https://user-images.githubusercontent.com/60759137/153909607-d6f8f634-a2a4-4204-982a-42f0912a780e.png)
First time in with no frame

### Steps to Reproduce

1. Create an app with two pages in the shell's flyout layout
2. Create a page with a frame
3. Add some simple controls such as a label inside the frame
4. Run the program and observe the layout - It should be incorrectly drawn
5. Switch to the other page and then return to the first page - it will be correctly drawn
6. The same behavior is exhibited when the first page is shown and you simply resize the window.

### Version with bug

Preview 12 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows

### Affected platform versions

net6.0-windows10.0.19041, net6.0-andriod... I did not test with my iPhone, but my guess is the same behavior will be exhibited there...

### Did you find any workaround?

No.

### Relevant log output

```shell
N/A
```

**Comment by davidortinau** [comment]
Thanks Steve. It may be fixed for you, but we certainly have more work to do here. This is the first run screen of the WeatherTwentyOne app for Windows.

<img width="1431" alt="image" src="https://user-images.githubusercontent.com/41873/154607388-1a88f638-8cd7-4f00-90c2-5b0ef7a1906c.png">

* The Flyout is not sized to the width provided
* The Border controls in the right widget panel are not drawing the background 

Resizing the window does NOT fix it. I have to navigate away and back. Then I see:

<img width="1368" alt="image" src="https://user-images.githubusercontent.com/41873/154607517-fbd8a970-5d25-4ad7-8493-0396b65c38c2.png">


Now I can see that the Border controls in the right panel are cut off on the right side. Frame worked, but Border is the problem here.

https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Views/WidgetsPanel.xaml#L8-L41

**Comment by mobycorp** [comment]
Hi David,

When I ran the weather app last week, I did observe the flyout acting as
you described.  When the app is first displayed the flyout is wonky as your
screenshot shows.  Then, as I indicated in my write-up when the page is
changed, the layout is invalidated and recalculated and the layout looks
"correct".  I say that in quotes because I know it looked nothing like the
screenshots in Github.  The details portion on the right-hand side of the
display never looked like your screenshot.  When I installed preview 13
yesterday, I saw that the invalid layout I was getting at the content
inside a frame appeared to be fixed.  I did try a workaround using a
border, but it didn't work for me as I had hoped so I didn't pursue it
further.

Is there anything I can try in my investigation to help?  I'm excited to be
this close to the action!
Steve

**Comment by Fabi** [comment]
Any news regarding that?

**Comment by wenwen60** [comment]
Reproduce on windows using VS2022(17.2.0 Preview 2.0), attach repro sample
[4672.zip](https://github.com/dotnet/maui/files/8159113/4672.zip).

**Comment by MaxtorCoder** [comment]
Any news regarding this issue?

**Comment by samhouts** [comment]
On 6.0.400-preview.1.5782, the Flyout is sized properly, but I still see cropping on the Border (see https://github.com/dotnet/maui/issues/4672#issuecomment-1043782945).

**Comment by mobynet1** [comment]
I reported this issue a year and a half ago and it is still an issue. Now, I am seeing this with the flyout menu. First time in, the flyout menu items are not spaced correctly and then when you tweak the width of the overall window, things are drawn correctly.

First time in:
![Screenshot 2023-09-15 110039](https://github.com/dotnet/maui/assets/124538071/b63cd449-aec7-4a28-98ec-06797de68abb)

Pull on an edge to resize the window:
![Screenshot 2023-09-15 110112](https://github.com/dotnet/maui/assets/124538071/1623d567-3f27-4300-b66d-c676aca4b6c0)

As I said, I reported this a LOOONG time ago and nothing has been done. In fact, it looks like the cancer is spreading. How are we supposed to develop world-class applications using MAUI when they don't render correctly and basically look like crap.

**Comment by erikbloodshed** [comment]
Don't use this framework. Microsoft is getting all it's developer to focus on the cloud and its AI bullshits! This project is dead already. Thank you.

**Comment by Fabi** [comment]
Still broken yay

**Comment by jeremy-visionaid** [comment]
Yeah, this one has been bugging me for years too... Still present in net9.0 / 9.0.10 SR1 :(

**Comment by jeremy-visionaid** [comment]
@kubaflo I believe this one was fixed some time ago in v10.0.x. Works for me at least, thanks!

**Comment by kubaflo** [comment]
Thanks @jeremy-visionaid — closing as fixed on the strength of your retest on v10.0.x.

For transparency: that confirmation comes from a community member rather than the original reporter. @SecureSharing, if the first-draw layout problem still reproduces for you on .NET 10, just comment here and this will be reopened — there's no need to file a new issue.

> Disclosure: this comment was generated by AI.
